src/vs/workbench/contrib/positronConsole/browser/components/consoleInstanceInfoButton.tsx

```diff
@@ -0,0 +1,135 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2025 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// CSS.
+import './consoleInstanceInfoButton.css';
+
+// React.
+import React, { useEffect, useRef, useState } from 'react';
+
+// Other dependencies.
+import { localize } from '../../../../../nls.js';
+import * as DOM from '../../../../../base/browser/dom.js';
+import { PositronModalPopup } from '../../../../browser/positronComponents/positronModalPopup/positronModalPopup.js'
+import { ActionBarButton } from '../../../../../platform/positronActionBar/browser/components/actionBarButton.js';
+import { PositronModalReactRenderer } from '../../../../browser/positronModalReactRenderer/positronModalReactRenderer.js';
+import { usePositronConsoleContext } from '../positronConsoleContext.js';
+import { PositronButton } from '../../../../../base/browser/ui/positronComponents/button/positronButton.js';
+import { ILanguageRuntimeSession } from '../../../../services/runtimeSession/common/runtimeSessionService.js';
+import { DisposableStore } from '../../../../../base/common/lifecycle.js';
+
+const positronConsoleInfo = localize('positron.console.info.label', "Console information");
+const showKernelOutputChannel = localize('positron.console.info.showKernelOutputChannel', "Show Kernel Output Channel");
+
+export const ConsoleInstanceInfoButton = () => {
+	// Hooks.
+	const positronConsoleContext = usePositronConsoleContext();
+
+	// Reference hooks.
+	const ref = useRef<HTMLButtonElement>(undefined!);
+
+	const handlePressed = () => {
+		if (!positronConsoleContext.activePositronConsoleInstance) {
+			return;
+		}
+
+		// Create the renderer.
+		const renderer = new PositronModalReactRenderer({
+			keybindingService: positronConsoleContext.keybindingService,
+			layoutService: positronConsoleContext.workbenchLayoutService,
+			container: positronConsoleContext.workbenchLayoutService.getContainer(DOM.getWindow(ref.current)),
+			parent: ref.current
+		});
+
+		renderer.render(
+			<ConsoleInstanceInfoModalPopup
+				anchorElement={ref.current}
+				renderer={renderer}
+				session={positronConsoleContext.activePositronConsoleInstance.session}
+			/>
+		);
+	}
+
+	// Render.
+	return (
+		<ActionBarButton
+			iconId='info'
+			align='right'
+			tooltip={positronConsoleInfo}
+			ariaLabel={positronConsoleInfo}
+			onPressed={handlePressed}
+			ref={ref}
+		/>
+	)
+};
+
+interface ConsoleInstanceInfoModalPopupProps {
+	anchorElement: HTMLElement;
+	renderer: PositronModalReactRenderer;
+	session: ILanguageRuntimeSession;
+}
+
+const ConsoleInstanceInfoModalPopup = (props: ConsoleInstanceInfoModalPopupProps) => {
+	const [sessionState, setSessionState] = useState(props.session.getRuntimeState());
```

Comment:
Throwing back to a [React tip from Nick](https://github.com/posit-dev/positron/pull/2899#pullrequestreview-2028647997) -- we can avoid calling `getRuntimeState()` on every re-render by wrapping the function call in an arrow function:

```suggestion
	const [sessionState, setSessionState] = useState(() => props.session.getRuntimeState());
```

(re-commented as the original suggestion looked to be on the wrong line, not sure how that happened!)

Reply:
Thank you for pointing that out! I've gone ahead and made this change. I also updated the debug session component to have the same improvements.